configure.ac

```diff
@@ -376,12 +376,50 @@ fi
 
 #fi
 
-# What SVN revision are we using?
-define([svnversion], esyscmd([sh -c "svnversion -n"]))dnl
-if test "x$want_cuda" != "xUnversioned directory"; then
- AC_DEFINE(CISTEM_SVN_REV, "svnversion", [SVN Revision])
+# What SVN revision are we using? Inner if clause will not work if svnversion is not callable
+AC_CHECK_PROG(SVB_CHECK,svnversion,yes)
+if test "x$SVN_CHECK" == "xyes" ; then
+  define([svnversion], esyscmd([sh -c "svnversion -n"]))dnl
+  if test "x$svnversion" != "xUnversioned directory"; then
+  AC_DEFINE(CISTEM_SVN_REV, "svnversion", [SVN Revision])
+  fi
 fi
 
+if test x$compile_experimental = xtrue ; then
+
+AC_CHECK_PROG(GIT_CHECK,git,yes)
+AC_CHECK_FILE(/groups/himesb/git/cisTEM_downstream_bah/.git/HEAD,
+  [
+    if test "x$GIT_CHECK" == "xyes"; then 
+      # Get the current branch, which could have multiple refs, which is why we need the awk (*) marks the active branch.
+      AC_SUBST([CISTEM_CURRENT_BRANCH],[`git  branch --points-at  HEAD | awk '/^\*/{print $2}'`])
+      # Get the first reachable tag - number of commits since that tag - short commit hash
+      AC_SUBST([CISTEM_VERSION_tag_nCommits_commitish],[`git describe --tags`])
+      # Get both an easily parsed date, as well as a more readable date
+      AC_SUBST([CISTEM_TIME_YYMMDDHHMMSS],[`git show --no-patch --no-notes --pretty='%cd' --date=format:%y%m%d%H%M%S`])
+      AC_SUBST([CISTEM_TIME_READABLE],[`git show --no-patch --no-notes --pretty='%ci' --date=format:%y%m%d%H%M%S`]) 
+      # Get the git configured user name (I believe this is required to push up to the repo)
+      AC_SUBST([CISTEM_GIT_USERNAME],[`git config --get user.name`])
+
+      # Print out what we found
+      AC_MSG_NOTICE([CISTEM_CURRENT_BRANCH is $CISTEM_CURRENT_BRANCH])
+      AC_MSG_NOTICE([CISTEM_VERSION_tag_nCommits_commitish is $CISTEM_VERSION_tag_nCommits_commitish])
+      AC_MSG_NOTICE([CISTEM_TIME_YYMMDDHHMMSS is $CISTEM_TIME_YYMMDDHHMMSS])
+      AC_MSG_NOTICE([CISTEM_TIME_READABLE is $CISTEM_TIME_READABLE])
+      AC_MSG_NOTICE([CISTEM_GIT_USERNAME is $CISTEM_GIT_USERNAME])
+
+      
+      CXXFLAGS="$CXXFLAGS -DCISTEM_CURRENT_BRANCH=\"\\\"${CISTEM_CURRENT_BRANCH}\\\"\" -DCISTEM_VERSION_tag_nCommits_commitish=\"\\\"${CISTEM_VERSION_tag_nCommits_commitish}\\\"\" -DCISTEM_TIME_YYMMDDHHMMSS=\"\\\"${CISTEM_TIME_YYMMDDHHMMSS}\\\"\" -DCISTEM_TIME_READABLE=\"\\\"${CISTEM_TIME_READABLE}\\\"\" -DCISTEM_GIT_USERNAME=\"\\\"${CISTEM_GIT_USERNAME}\\\"\""
+      CPPFLAGS="$CPPFLAGS -DCISTEM_CURRENT_BRANCH=\"\\\"${CISTEM_CURRENT_BRANCH}\\\"\" -DCISTEM_VERSION_tag_nCommits_commitish=\"\\\"${CISTEM_VERSION_tag_nCommits_commitish}\\\"\" -DCISTEM_TIME_YYMMDDHHMMSS=\"\\\"${CISTEM_TIME_YYMMDDHHMMSS}\\\"\" -DCISTEM_TIME_READABLE=\"\\\"${CISTEM_TIME_READABLE}\\\"\" -DCISTEM_GIT_USERNAME=\"\\\"${CISTEM_GIT_USERNAME}\\\"\""
+    else
+      AC_MSG_NOTICE([We DO NOT have git on path])
+      AC_DEFINE(CISTEM_VERSION_TEXT, "2.0.0-alpha", [A hard coded version moved in from defines.h]) 
```

Comment:
Isn't this PR all about creating 1.1? Shouldn't that be what it reads here?

Reply:
This commit only makes changes that appear under the control of "--enable-experimental" configure option. I'm not sure what we want to put for "official" releases, so I have defaulted to leave the version text that was already in place.

Reply:
What exactly is the output of "git describe --tags --dirty"  I get 1.0.0-566-g74d697f  - This means it is the 1.0.0 branch with 566 additional commits right?    

I think that after the merge the master should say 2.0.0 and CISTEM_VERSION_TEXT should say 2.0.0-alpha.  The 1.1 release should say 1.1.0 and CISTEM_VERSION_TEXT should say 1.1.0-tm-beta.  I am not sure what the best way to accomplish this is though?  

Is the 1.1 being retained as a branch or a tag?

Reply:
## CISTEM_VERSION_TEXT 
Currently CISTEM_VERSION_TEXT will say 2.0.0-alpha on both branches, **_if_** --enable-experimental is not configured. If we pull that requirment, then all we need to do is to place an annotated tag on the master branch named 2.0.0-alpha, but this would need to be in a separate PR.
## Other info
That is no problem. The remaining question then, is do you want the any/all of the other info to only show up with --enable-experimental configured or all the time?
- Compiled by: <name>
- Commit datetime
- From branch

## alpha/beta
The main purpose for 1.1.0 is for reproducibility of the eLife paper. As there is no plan for further development of new functionality on the 1.1.0 branch, I'm not sure adding "beta" is needed. This branch is to persist as a "version of record" that only receives critical bug updates, relevant to running the code (i.e. preventing crashes.)

Reply:
The reason i would like to add beta is otherwise it sounds like it is stable, which it most definitely is not.  In fact, it should not be used over the 1.0.0-beta release for single particle.  Maybe we should call it 1.1.0-template-matching or something?

I am happy for commit datetime and from branch to be included all the time, i don't think we need compiled by though.

it sounds like this should be merged into the master, then you should commit to the branch so that the CISTEM_VERSION_TEXT says the correct thing when compiled without --enable-experimental.